store/versioned_cachingstore.go

```diff
@@ -371,11 +370,23 @@ func (c *versionedCachingStore) SaveVersion() ([]byte, int64, error) {
 	return hash, version, err
 }
 
-func (c *versionedCachingStore) GetSnapshot() Snapshot {
-	return newVersionedCachingStoreSnapshot(
-		c.VersionedKVStore.GetSnapshot(),
-		c.cache, c.version-1, c.logger,
-	)
+func (s *versionedCachingStore) GetSnapshot() Snapshot {
+	snapshot, err := s.GetSnapshotAt(0)
+	if err != nil {
+		panic(err)
+	}
+	return snapshot
+}
+
+func (c *versionedCachingStore) GetSnapshotAt(version int64) (Snapshot, error) {
+	if version == 0 {
+		return newVersionedCachingStoreSnapshot(
+			c.VersionedKVStore.GetSnapshot(),
+			c.cache, c.version-1, c.logger,
+		), nil
+	} else {
+		return c.VersionedKVStore.GetSnapshotAt(version)
```

Comment:
Why doesn't this use `newVersionedCachingStoreSnapshot`?

Reply:
It could do, but the version caching wrapper did not seem necessary for a read-only store used for a replay.